src/NServiceBus.Core.Tests/Unicast/RunningEndpointInstanceTest.cs

```diff
@@ -1,28 +1,50 @@
 ﻿namespace NServiceBus.Unicast.Tests
 {
+    using System;
     using System.Threading.Tasks;
     using NUnit.Framework;
     using Settings;
 
     [TestFixture]
     public class RunningEndpointInstanceTest
     {
-        [Test]
-        public async Task ShouldAllowMultipleStops()
+        static RunningEndpointInstance Create()
         {
             var settings = new SettingsHolder();
 
-            var testee = new RunningEndpointInstance(
+            var testInstance = new RunningEndpointInstance(
                 settings,
                 new HostingComponent(null, true),
                 null,
                 new FeatureComponent(settings),
                 new MessageSession(new FakeRootContext()),
                 null);
+            return testInstance;
+        }
+
+        [Test]
+        public async Task ShouldAllowMultipleStops()
+        {
+            var testInstance = Create();
+
+            await testInstance.Stop();
+
+            Assert.That(async () => await testInstance.Stop(), Throws.Nothing);
+        }
+
+        [Test]
+        public async Task ShouldThrowExceptionAfterInvokingStop()
```

Comment:
Maybe it's just me, but I would have probably split these up into dedicated tests, but it's really a preference.